woocommerce/woocommerce
# Adds tests to monitor number of JS files included in pages

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





After the 8.5.x release, there was some discussion around "bloat" and how the number of included JS files kept climbing.  While not an immediate problem, an ever-increasing number of JS inclusions can lead to performance issues.

This PR introduces an e2e test allowing for pages to be added along with the number of expected JS includes for that page.  This data has been split out, so that it can be easily updated -- but will require an intentional update to the test if new JS includes are to be introduced.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. `pnpm env:restart`
2. `pnpm test:e2e-pw js-file-monitor/`
3. Ensure tests pass in CI



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
